docs/src/main/asciidoc/qute-reference.adoc

```diff
@@ -242,6 +242,47 @@ If you need to render the unescaped value:
 <1> `title` that resolves to `Expressions & Escapes` will be rendered as `Expressions &amp;amp; Escapes`
 <2> `paragraph` that resolves to `<p>My text!</p>` will be rendered as `<p>My text!</p>`
 
+===== Virtual Methods
+
+A virtual method is a part of an expression that looks like a regular Java method invocation.
+It's called "virtual" because it does not have to match the actual method of a Java class.
+In fact, any virtual method is handled by a value resolver.
+The only difference compared to regular parts of an expression is that for a virtual method a value resolver accepts parameters that are also expressions.
+Parameters are always evaluated first.
+
+.Virtual Method Example
+[source,html]
+----
+<html>
+<h1>{item.buildName(item.name,5)}</h1> <1>
+</html>
+----
+<1> `buildName(item.name,5)` represents a virtual method with name `buildName` and two parameters: `item.name` and `5` (integer literal). 
```

Comment:
It would be nice to add a snippet for the `buildName` virtual method (much easier than explaining it) :smile:

Reply:
Ok. I'dd add some more description.